### PR TITLE
feature: 034 Create change password page 

### PR DESCRIPTION
**Pull request by halfangel37**
## Change description
I create a feature change password and it place inside the user profile , so I created a folder name 'profile' to contain all needed file( components, views, store) , I do it with my experience with register screen, the form also have validated just not like in the design but it familiar with register screen , validate make it easier to deal with the errors, the css of the dashboard created by Thanh Nguy make my form not in the center when responsive with mobile device, so I have to add some css to it. Hope you review it soon.

## Demo 📺
https://www.loom.com/share/9ad4aaa1e8d94ff1aff1ccbe07f860fb
Update Demo after resolve conversations:
https://www.loom.com/share/89c4a23c73284337b1e027413c0a01be

### CODE REVIEW

File: src/http-common.js [hunk 1]
```diff
@@ -3,6 +3,6 @@ import axios from "axios";
 export const HTTP = axios.create({
   baseURL: `http://intern-2021.southeastasia.cloudapp.azure.com:4000/api/v1`,
   headers: {
-    Authorization: "Bearer {token}",
+    Authorization: `Bearer ${localStorage.token}`,
   },
```
Comment (truonghungit) [inline src/http-common.js:7]: Wait for [#044](https://trello.com/c/Lhh5tkRq/46-044-creating-http-clients-using-axios-instances), then we will have an HTTP interceptor that adds authorization token to the header auto

File: src/profile/store/Profile.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+export default {
+  namespaced: true,
+  state: {
+    changePasswordMessage: "",
+  },
+  mutations: {
+    SET_PASSWORD_MESSAGE(state, errorMessage) {
+      state.changePasswordMessage = errorMessage;
+
+      console.log("state:", state.changePasswordMessage);
```
Comment (truonghungit) [inline src/profile/store/Profile.js:12]: Remove console log

File: src/profile/views/Profile.vue [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  components: { ChangePassword },
+  data: () => ({ isActive: true }),
+
+  methods: {
+    changePassword(value) {
+      this.$store.dispatch("profile/changePassword", {
+        oldPassword: value.currentPassword,
+        newPassword: value.newPassword,
+      });
+    },
```
Comment (truonghungit) [inline src/profile/views/Profile.vue:38]:
```suggestion
    changePassword({currentPassword, newPassword}) {
      this.$store.dispatch("profile/changePassword", {
        oldPassword: currentPassword,
        newPassword,
      });
    },
```

File: src/profile/views/Profile.vue [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    displayForm() {
+      this.isActive = false;
+    },
+    undisplayForm() {
+      this.isActive = true;
+    },
+  },
```
Comment (truonghungit) [inline src/profile/views/Profile.vue:45]: ?? Please check [tabs document](https://vuetifyjs.com/en/components/tabs/)

File: src/profile/views/Profile.vue [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+}
+.tabs {
+  margin-bottom: 30px;
+}
+#form-container {
+  padding: 0;
+}
+.active {
+  display: none;
+}
```
Comment (truonghungit) [inline src/profile/views/Profile.vue:66]: active but display `none` ??

File: src/store.js [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
+import profile from "./profile/store/Profile";
 import * as signin from "@/signin/store/signin.js";
 import * as signout from "@/signout/store/signout.js";
 Vue.use(Vuex);
 export default new Vuex.Store({
   state: {},
   mutations: {},
   actions: {},
-  modules: { register, signin, signout },
+  modules: { register, signin, signout, profile },
```
Comment (truonghungit) [inline src/store.js:12]:
```suggestion
  modules: { 
         register,
         signin,
         signout,
         profile,
   },
```

Comment (truonghungit) [reply src/store.js:12]: This help us avoid conflict when merging code

File: src/store.js [hunk 7]
```diff
@@ -1,12 +1,13 @@
 import Vue from "vue";
 import Vuex from "vuex";
 import register from "./register/store/Register";
+import profile from "./profile/store/Profile";
```
Comment (truonghungit) [inline src/store.js:4]: Rename file `./profile/store/Profile` to `./profile/store/index.js` then we can import `@/profile/store`
```suggestion
import profile from "@/profile/store";
```

File: src/profile/components/ChangePassword.vue [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+h3 {
+  color: #3e4c59;
+  margin-bottom: 1.5rem;
+}
+.form {
+  box-shadow: 0px 2px 6px rgba(0, 0, 0, 0.16);
+  border-radius: 4px;
+  padding: 1.5rem;
+}
+</style>
```
Comment (truonghungit) [inline src/profile/components/ChangePassword.vue:104]: Can you rename this component to `ChangePasswordForm`

File: src/profile/components/ChangePassword.vue [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    ref="changePasswordForm"
+    v-model="formValidity"
+  >
+    <v-alert v-if="changePasswordMessage" type="error">
+      {{ changePasswordMessage }}
+    </v-alert>
+    <v-alert v-if="changePasswordMessage === null" type="success">
+      Congratulations ! You password are changed successfully !
+    </v-alert>
+
```
Comment (truonghungit) [inline src/profile/components/ChangePassword.vue:14]: ??

**Review by truonghungit — CHANGES_REQUESTED** [review]

File: src/profile/components/ChangePassword.vue (hunk 9 above)
Comment (truonghungit) [reply src/profile/components/ChangePassword.vue:14]: 1. You can move alerts into `Profile.vue ` therefore, you don't need to pass the `changePasswordMessage` props
2. If you want to show  alerts in here, you can rename this props to `errorMessage` for more clear

**Review by truonghungit — CHANGES_REQUESTED** [review]

File: src/profile/store/index.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+            commit(
+              "SET_PASSWORD_MESSAGE",
+              error.response.data.errors[0].message
+            );
+          } else {
+            commit(
+              "SET_PASSWORD_MESSAGE",
+              error.response.data.errors.NewPassword[0]
+            );
+          }
```
Comment (truonghungit) [inline src/profile/store/index.js:33]:
```suggestion
           const message = error.response.data.errors[0] 
                   ? error.response.data.errors[0].message
                   :  error.response.data.errors.NewPassword[0];
                   
           commit( "SET_PASSWORD_MESSAGE", message);
```

File: src/profile/views/Profile.vue [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+      color="#4f2566"
+      slider-color="#E7AC28"
+      slider-size="3"
+    >
+      <v-tab v-for="item in items" :key="item">
+        {{ item }}
+      </v-tab>
+    </v-tabs>
+    <v-tabs-items v-model="tab">
+      <v-tab-item v-for="item in items" :key="item">
```
Comment (truonghungit) [inline src/profile/views/Profile.vue:17]: Nice work

**Review by truonghungit — CHANGES_REQUESTED** [review]
Nice work

**Review by truonghungit — APPROVED** [review]

**Review by luantc13 — APPROVED** [review]
Great work!

**Pull request merged**